var/spack/repos/builtin/packages/snappy/package.py

```diff
@@ -24,6 +24,17 @@ class Snappy(CMakePackage):
 
     patch("link_gtest.patch", when="@:1.1.8")
 
+    patch(
```

Comment:
I thought that the url of the patch in the code answers all the questions: the patch is fetched from the upstream repo, why it is needed is described in the [commit message](https://github.com/google/snappy/commit/8dd58a519f79f0742d4c68fbccb2aed2ddb651e8) of the patch and from the description it's more or less clear that it makes it possible to build the package.

Fine, I'll try to come up with something :)

Reply:
I have also removed the compiler constraint for the patch because it might be needed for other compilers/targets as well and should not break the existing working cases.